### PR TITLE
*: make schema as a pointer.

### PR DESCRIPTION
**Pull request by hanfei1991**
Make Schema as pointer and change the interface of schema.
@shenli @coocood @zimulala @tiancaiamao PTAL

### CODE REVIEW

File: plan/logical_plan_builder.go [hunk 1]
```diff
@@ -1170,6 +1171,6 @@ func (b *planBuilder) buildDelete(delete *ast.DeleteStmt) LogicalPlan {
 	del.self = del
 	del.initIDAndContext(b.ctx)
 	addChild(del, p)
+	del.SetSchema(expression.NewSchema())
```
Comment (zimulala) [inline plan/logical_plan_builder.go]: Why add this line of code now？

**Comment by shenli** [comment]
What's the benefit of this PR?

File: plan/logical_plan_builder.go (hunk 1 above)
Comment (hanfei1991) [inline plan/logical_plan_builder.go]: the `adapter.go` will check `schema.Len`, we cannot let it be a nil pointer.

**Comment by hanfei1991** [comment]
@shenli  In future, we may add more information to schema structure, so we'd better make it a pointer to avoid mem allocating.

**Comment by zimulala** [comment]
LGTM

**Comment by shenli** [comment]
LGTM

**Pull request merged**